codec_test.go

```diff
@@ -105,6 +105,40 @@ func TestV2Codec(t *testing.T) {
 	}
 }
 
+func Test_setting_withDecodeOnly_disables_encoding(t *testing.T) {
+	d := &memory.Driver{}
+	srv := httptest.NewServer(server.NewHttpHandler(d))
+	defer srv.Close()
+	decodeOnlyCodec := setUpWithServer(t, "v2", srv, true)
+
+	defaultCodec := setUpWithServer(t, "v2", srv, false)
+
+	payload := common.Payload{
+		Metadata: map[string][]byte{
+			"foo":                     []byte("bar"),
+			"baz":                     []byte("qux"),
+			"remote-codec/key-prefix": []byte("1234"),
+		},
+		Data: []byte("this is a longer message blah blah blah blah blah blah blah"),
+	}
+
+	resp1, err := decodeOnlyCodec.Encode([]*common.Payload{&payload})
+	if err != nil {
```

Comment:
there is no `if err != nil` needed here. The `require` should be executed all the times. If `err` is not `nil` it will fail the test, otherwise it passes. The opposite is `require.Error`

Same for the others.

Reply:
you're right fixed :)